packages/client/TODO.md

```diff
@@ -0,0 +1,182 @@
+
+- [AM] client.profile.createSetProfileImageURITypedData(request)
+- [AM] client.profile.createSetProfileImageURIViaDispatcher(request)
+
+- [AM] client.profile.createBurnProfileTypedData(request)
+- [AM] client.profile.createSetDefaultProfileTypedData(request)
+- [AM] client.profile.createSetDispatcherTypedData(request)
+
+- [AM] client.profile.addInterests(request)
+- [AM] client.profile.removeInterests(request)
+
+### Follow profile
+
+- [AM] client.profile.createFollowTypedData(request)
+- [AM] client.profile.createUnfollowTypedData(request)
+- [AM] client.profile.createSetFollowModuleTypedData(request)
+- [AM] client.profile.createSetFollowNFTUriTypedData(request)
+- [APQ] client.profile.pendingApprovalFollows(request)
+
+- [Q] client.profile.doesFollow(request)
+- [PQ] client.profile(s).following({ address })
+- [PQ] client.profile.fetchFollowers({ profileId })
+- [Q] client.profile.followerNftOwnedTokenIds({ address, profileId })
+
+## Publication(s)
+
+### Query publications
+
+- [Q] client.publication.fetch(request) or client.publications.fetchOne()
+- [PQ] client.publication.fetchAll(request)
+- [Q] client.publication.validateMetadata(request)
+- [PQ] client.publication.whoCollected({ publicationId })
+
+- [PQ] client.publication(s).forSale({ profileId })
+- [Q] client.publication.metadataStatus(request)
+
+### Create or edit publications
+
+- [AM] client.publication.createPostTypedData(request)
+- [AM] client.publication.createPostViaDispatcher(request)
+
+- [AM] client.publication.createCommentTypedData(request)
+- [AM] client.publication.createCommentViaDispatcher(request)
+
+- [AM] client.publication.createMirrorTypedData(request)
+- [AM] client.publication.createMirrorViaDispatcher(request)
+
+- [AM] client.publication.hide({ publicationId })
+- [M] client.publication.createAttachMediaData(request)
+
+- gated publications TBD
+
+## ProtocolStats
+
+- [Q] client.protocolStats.fetch(request)
+
+## UserSigNonces
+
+- [AQ] client.nonces.fetch()
+
+## Proxy actions
+
+- [AM] client.proxyActions.freeFollow({ profileId })
+- [AM] client.proxyActions.freeCollect({ publicationId })
+- [AQ] client.proxyActions.checkStatus({ proxyActionId })
+
+## Reaction(s)
+
+- [x] [AM] client.reaction.add(request)
+- [x] [AM] client.reaction.remove(request)
+- [PQ] client.reaction(s).toPublication({ publicationId })
+
+## Reporting
+
+- [AM] client.report.publication({ publicationId, reason, additionalComments })
+
+## Revenue
+
+- [PQ] client.revenue.profilePublications({ profileId, limit })
+- [Q] client.revenue.profileFollow({ profileId })
+- [Q] client.revenue.publication({ publicationId })
+
+## Search
+
+- [PQ] client.search.profiles({ query, limit })
+- [PQ] client.search.publications({ query, limit })
+
+## Feed
+
+- [PQ] client.feed.fetch({ profileId })
+
+## Other considerations
+
+- paginated response helper
+- more helpers like `isValidHandle`
+- how to split a group that has both authenticated and public requests?
```

Comment:
I reckon authenticated vs. not-authenticated should be a reason to split into 2 separate sub-modules.

Also bear in mind some queries require authentication but conceptually they might not need to. I know about `enabledCurrencies` but there might be others. Maybe we can ask to remove such constraint at the API level.

Reply:
I'm thinking more about checking if the client is authenticated on a method call level, not a module level. That makes the interface nicer and more consistent.

look at Reactions (from TODO)

```
[A] authenticated
[P] paginated
[M] mutation
[Q] query
 
- [AM] client.reactions.add(request)
- [AM] client.reactions.remove(request)
- [PQ] client.reactions.toPublication({ publicationId })
```

it would be confusing to have
```
client.authReactions.add()
// and
client.publicReactions.toPublication()
```

nicer approach (if not auth)
```
client.reactions.toPublication() // all good
client.reactions.add(request) // throw NotAuthenticatedError()

// then
client.refreshToken()
client.reactions.add(request)  // all good now
```

Reply:
thinking out loud, we had a talk with @cesarenaldi to expose our `Result` "monad" to the external world to make it more typesafe, maybe we should follow the same convention for the sdk and return `Result` from methods? 
Thoughts?

Reply:
I created a ticket to think about it later, for now, I moved throwing of NotAuthenticatedError to a method call level, see Reactions.ts
https://avara.height.app/lens-sdk/T-5676